ssh_key_rotator/publisher/key_publisher.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractmethod
+import os
+import shutil
+import sys
+import boto3
+from ssh_key_rotator.custom_keygen import (
+    generate_private_public_key_in_file,
+    generate_private_public_key,
+)
+from ssh_key_rotator.util import get_user_path, get_username
+
+
+class Publisher(ABC):
+    """Abstract key publisher base class"""
+
+    @abstractmethod
+    def publish_new_key(self) -> str:
+        """Abstract method for publishing a new public key"""
+
+
+class S3Publisher(Publisher):
+    """S3 Publisher class"""
+
+    def __init__(self, bucket_name: str, host: str, user_id: str):
+        self.bucket_name = bucket_name
+        self.host = host
+        self.user_id = user_id
+
+    def publish_new_key(self) -> str:
+        # Generate new public/private key pair
+        private_key, public_key = generate_private_public_key()
+
+        # Store the new public key in S3 bucket
+        s3_client = boto3.client("s3")
+        s3_client.put_object(
+            Body=public_key,
+            Bucket=self.bucket_name,
+            Key=f"{self.host}/{self.user_id}-cert.pub",
+        )
+
+        # Store the private key on the local machine
+        private_key_path = f"{get_user_path()}/.ssh/{self.host}/{self.user_id}"
+        with open(private_key_path, "wb") as private_out:
+            private_out.write(private_key)
+        shutil.chown(private_key_path, user=get_username(), group=-1)
+        os.chmod(private_key_path, 0o600)
+
+        return public_key.decode()
+
+
+class LocalPublisher(Publisher):
+    """Local Publisher class"""
+
+    def __init__(self, host: str, user_id: str):
+        self.host = host
+        self.user_id = user_id
+
+    def publish_new_key(self) -> str:
```

Comment:
what happens with the private key?

Reply:
Gets stored on the host machine via the generate method